pkg/cloud/scope/clients.go

```diff
@@ -129,8 +131,8 @@ func NewIAMClient(scopeUser cloud.ScopeUsage, session cloud.Session, target runt
 }
 
 // NewSTSClient creates a new STS API client for a given session
-func NewSTSClient(scopeUser cloud.ScopeUsage, session cloud.Session, target runtime.Object) stsiface.STSAPI {
-	stsClient := sts.New(session.Session())
+func NewSTSClient(scopeUser cloud.ScopeUsage, session cloud.Session, logger logr.Logger, target runtime.Object) stsiface.STSAPI {
+	stsClient := sts.New(session.Session(), aws.NewConfig().WithLogLevel(getAWSLogLevel(logger)))
```

Comment:
If we don't use the a handler approach I think  we should also use [WithLogger](https://github.com/aws/aws-sdk-go/blob/c30aae45d1dde08973dc872c5b2e81b99c5ac89a/aws/config.go#L365) so that we can use the logger that's already being used for the rest of the log entries in capa.  WDYT?

Reply:
Yes, that totally makes sense! Let me fix that!

Reply:
I've tried using the logger with `WithLogger` but found it really difficult to implement. So now I'm trying to reimplement it using a handler so give me some more time 🙇

Reply:
You could probably do something like this (I've not tested this much):

```go
aws.NewConfig().WithLogLevel(aws.LogDebugWithHTTPBody).WithLogger(WrapLogr(log))

func WrapLogr(logger logr.Logger) aws.Logger {
	return &LogrWrapper{
		log: logger,
	}
}

type LogrWrapper struct {
	log logr.Logger
}

func (l *LogrWrapper) Log(msgs ...interface{}) {
	switch len(msgs) {
	case 0:
		return
	case 1:
		l.log.Info(msgs[0].(string))
	default:
		l.log.Info(msgs[0].(string), msgs[:1]...)
	}
}
```

Reply:
Thank you for your guidance! Let me try that 🙇

Reply:
Fixed it and the log looks like the one below!

```
I0207 06:17:36.100798      34 logs.go:40] controllers/AWSMachine "msg"="DEBUG: Request ec2/DescribeInstances Details:\n---[ REQUEST POST-SIGN ]-----------------------------\nPOST / HTTP/1.1\r\nHost: ec2.ap-northeast-1.amazonaws.com\r\nUser-Agent: aws.cluster.x-k8s.io/ aws-sdk-go/1.36.26 (go1.14.5; linux; amd64)\r\nContent-Length: 76\r\nAuthorization: AWS4-HMAC-SHA256 Credential=AKIA47GPXOIYQK7YYYPC/20210207/ap-northeast-1/ec2/aws4_request, SignedHeaders=content-length;content-type;host;x-amz-date, Signature=343ff3b6d8cf7c4da25e7983ddd47e910d71de4b780cf999ad4680655bd218f5\r\nContent-Type: application/x-www-form-urlencoded; charset=utf-8\r\nX-Amz-Date: 20210207T061736Z\r\nAccept-Encoding: gzip\r\n\r\nAction=DescribeInstances\u0026InstanceId.1=i-0b453a09eca5e4624\u0026Version=2016-11-15\n-----------------------------------------------------" "awsMachine"="capi-cluster-api-provider-aws-md-0-qddkg" "cluster"="capi-cluster-api-provider-aws" "machine"="capi-cluster-api-provider-aws-md-0-5b97fcdb4c-v24gn" "namespace"="capa-system" 
```